libdnf/dnf-transaction.cpp

```diff
@@ -1432,16 +1432,8 @@ dnf_transaction_commit(DnfTransaction *transaction, HyGoal goal, DnfState *state
         goto out;
 
     // FIXME get commandline
-    if (sack) {
-        rpmdb_begin = dnf_sack_get_rpmdb_version(sack);
-    } else {
-        // if sack is not available, create a custom instance
-        rpmdb_version_sack = dnf_sack_new();
-        dnf_sack_load_system_repo(rpmdb_version_sack, nullptr, DNF_SACK_LOAD_FLAG_NONE, nullptr);
-        rpmdb_begin = dnf_sack_get_rpmdb_version(rpmdb_version_sack);
-        g_object_unref(rpmdb_version_sack);
-    }
-    swdb->beginTransaction(_get_current_time(), rpmdb_begin, "", priv->uid);
+    rpmdb_begin.reset(rpmdbCookie(rpmtsGetRdb(priv->ts)));
+    swdb->beginTransaction(_get_current_time(), rpmdb_begin.get(), "", priv->uid);
```

Comment:
In case that `rpmdb_end` will be `nullptr` it will end on crash in `std::string` constructor. `rpmdbCookie` can return nullptr when  failed.

Reply:
I fixed it.